docs/docsgen/source/technical/kv_cache.md

```diff
@@ -0,0 +1,21 @@
+<!--
+Copyright (c) ONNX Project Contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+(onnx-detail-kvcache)=
+
+# In-place KV Cache for Attention
+
+KV caching in attention-based models refers to a mechanism for storing previously computed Key and Value tensors during autoregressive generation. In decoder-only transformers, each new token must attend to all previous tokens using the attention mechanism. Normally, this would require recomputing the Key and Value projections for every prior token at each time step, which is inefficient. Instead, the KV cache stores these projections after they are first computed, allowing the model to reuse them for future tokens without recomputation. This significantly speeds up the generation process.
+
+Updating the KV cache in place means writing new Key and Value tensors directly into pre-allocated memory at the index corresponding to the current position in the sequence. This has several advantages: it avoids repeated memory allocation or copying, reducing computational overhead; it also allows better performance on hardware accelerators by enabling the use of fused kernels and reducing memory bandwidth usage. In-place updates are essential for achieving high throughput and low latency during inference, particularly for large language models deployed in real-time applications.
+
+ONNX opset-24 has introduced new features to faciliate the representation of in-place KV cache updates. This diagram shows an example use case:
+
+![InPlace KV Cache](InPlaceKVCache.png)
```

Comment:
Just a nit about the diagram: is this auto-generated? It seems to me that the layout would look better if the TensorScatter for past_k was higher and the one for past_v was lower ... with no need for edges crossing over.

Reply:
It's hand drawn actually.
I've updated the layout as you suggested.